In [1]:
import numpy as np
from scipy import stats

__Задача 1__

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

In [26]:
#Дано:
a_real = 17.5
disp = 4
std = disp**0.5
n = 100

In [27]:
samples = np.random.normal(a_real, std, size=n)
x_mean = samples.mean()
x_mean

17.526404521522444

Поскольку мы проверяем гипотезу относительно математического ожидания нормально распределённой случайной величины с известной дисперсией, будем использовать статистику
$$T(X) = \dfrac{\overline{X} - a}{\sigma / \sqrt{n}}$$

In [28]:
a = 17
t_x = (x_mean - a) / (sigma / n**0.5)
t_x

2.632022607612221

In [36]:
alpha = 0.025

In [33]:
ppf = stats.norm.ppf(1 - alpha / 2)
ppf

2.241402727604947

Критическая область справа:
$$(2.241, \infty).$$

В данном случае мы попали в критическую область (2.632 > 2.241), т.е. результат гипотезы отрицательный

__Задача 2__

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания `a` с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

In [69]:
std = 16
p = 0.95
n = 256
x_mean = 80
alpha = 1 - p
alpha

0.050000000000000044

In [70]:
t = stats.t.ppf(1 - alpha / 2, df=n - 1)
t

1.9693105698491928

In [71]:
#Доверительный интервал:
(x_mean - t * std / np.sqrt(n), x_mean + t * std / np.sqrt(n))

(78.0306894301508, 81.9693105698492)

__Задача 3__

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

In [73]:
samples = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
n = len(samples)
std = samples.std(ddof=1)
x_mean = samples.mean()
n, x_mean, std 

(10, 198.5, 4.453463071962462)

In [74]:
alpha = 0.01
t = stats.t.ppf(1 - alpha / 2, df=n - 1)
t

3.2498355440153697

In [75]:
#Доверительный интервал:
(x_mean - t * std / np.sqrt(n), x_mean + t * std / np.sqrt(n))

(193.92322840036354, 203.07677159963646)

Ответ - вверно, т.к. утверждаемый средний вес пачки печенья 200 г попадает в доверительный интервал выборки с вероятностью p = 99% 